storage/repair_queue_test.go

```diff
@@ -0,0 +1,155 @@
+	storePool := NewStorePool(g, TestTimeUntilStoreDeadOff, stopper)
+	a := makeAllocator(storePool)
+	replicateQ := makeReplicateQueue(g, a, clock)
+	repairQ := makeRepairQueue(storePool, &replicateQ, clock)
+
+	testCases := []struct {
+		replica        *Replica
+		storePoolAlive []proto.StoreID
+		storePoolDead  []proto.StoreID
+		eNeeds         bool
+		ePriority      int
+		eReplicas      []proto.Replica
+	}{
+		// Test 0: three stores, all alive, nothing to do
+		{
+			replica:        createTestReplicaForRepair([]proto.StoreID{1, 2, 3}),
+			storePoolAlive: []proto.StoreID{1, 2, 3},
+			storePoolDead:  []proto.StoreID{},
+		},
+		// Test 1: three stores, one dead store, repair the store
+		{
+			replica:        createTestReplicaForRepair([]proto.StoreID{1, 2, 3}),
+			storePoolAlive: []proto.StoreID{1, 2},
+			storePoolDead:  []proto.StoreID{3},
+			eNeeds:         true,
+			ePriority:      0,
+			eReplicas:      createReplicaSets([]proto.StoreID{3}),
+		},
+		// Test 2: three stores, two dead stores, can't repair due to not having quorum
+		{
+			replica:        createTestReplicaForRepair([]proto.StoreID{1, 2, 3}),
+			storePoolAlive: []proto.StoreID{1},
+			storePoolDead:  []proto.StoreID{2, 3},
+			ePriority:      1,
+			eReplicas:      createReplicaSets([]proto.StoreID{2, 3}),
+		},
+		// Test 3: five store, two dead stores, both should be repaired
+		{
+			replica:        createTestReplicaForRepair([]proto.StoreID{1, 2, 3, 4, 5}),
+			storePoolAlive: []proto.StoreID{1, 2, 3},
+			storePoolDead:  []proto.StoreID{4, 5},
+			eNeeds:         true,
+			ePriority:      0,
+			eReplicas:      createReplicaSets([]proto.StoreID{4, 5}),
+		},
+		// Test 4: five store, one dead stores, repair the store
+		{
+			replica:        createTestReplicaForRepair([]proto.StoreID{1, 2, 3, 4, 5}),
+			storePoolAlive: []proto.StoreID{1, 2, 3, 4},
+			storePoolDead:  []proto.StoreID{5},
+			eNeeds:         true,
+			ePriority:      -1,
+			eReplicas:      createReplicaSets([]proto.StoreID{5}),
+		},
+		// Test 5: three stores, all alive, but only 1 is in the store pool, nothing to do
+		{
+			replica:        createTestReplicaForRepair([]proto.StoreID{1, 2, 3}),
+			storePoolAlive: []proto.StoreID{1},
+			storePoolDead:  []proto.StoreID{},
+		},
+	}
+
+	for i, testCase := range testCases {
+		mockStorePool(storePool, testCase.storePoolAlive, testCase.storePoolDead)
+		aNeeds, aPriority, aReplicas := repairQ.needsRepair(testCase.replica)
+		if testCase.eNeeds != aNeeds {
+			t.Errorf("%d: Expected `needs` does not match actual. Expected: %t, Actual: %t", i, testCase.eNeeds, aNeeds)
+		}
+		if testCase.ePriority != int(aPriority) {
+			t.Errorf("%d: Expected `priority` does not match actual. Expected: %d, Actual: %f", i, testCase.ePriority, aPriority)
+		}
+		if !reflect.DeepEqual(testCase.eReplicas, aReplicas) {
+			t.Errorf("%d: Expected `replicas` does not match actual. Expected: %v, Actual: %v", i, testCase.eReplicas, aReplicas)
```

Comment:
tip: when comparing arrays, you generally want to put a newline on either side of each array. you should change this test so it fails and then look at the output, and you'll appreciate what that formatting does for future readers.